Lots of warnings during compilation on GNU/Linux

```
The Meson build system
Version: 0.61.2
Source dir: /home/kelson/code/libzim
Build dir: /home/kelson/code/libzim/build
Build type: native build
Project name: libzim
Project version: 8.1.0
C compiler for the host machine: ccache cc (gcc 11.3.0 "cc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0")
C linker for the host machine: cc ld.bfd 2.38
C++ compiler for the host machine: ccache c++ (gcc 11.3.0 "c++ (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0")
C++ linker for the host machine: c++ ld.bfd 2.38
Host machine cpu family: x86_64
Host machine cpu: x86_64
Checking for size of "off_t" : 8
Checking for size of "size_t" : 8
Found pkg-config: /usr/bin/pkg-config (1.8.0)
Run-time dependency liblzma found: YES 5.2.5
Run-time dependency libzstd found: YES 1.4.8
Run-time dependency xapian-core found: YES 1.4.19
Run-time dependency threads found: YES
Run-time dependency icu-i18n found: YES 70.1
Run-time dependency GTest found: YES 1.9.0
Configuring zim_config.h using configuration
Program libzim-compile-resources found: YES (/home/kelson/code/libzim/scripts/libzim-compile-resources)
Program download_test_data.py found: YES (/home/kelson/code/libzim/scripts/download_test_data.py)
Configuring config.h using configuration
Build targets in project: 33

Found ninja-1.8.2 at /home/kelson/bin/ninja
ninja: Entering directory `build'
[13/133] Generating static/resources with a custom command
stopwords/af ['stopwords', 'af']
stopwords/ar ['stopwords', 'ar']
stopwords/bg ['stopwords', 'bg']
stopwords/bn ['stopwords', 'bn']
stopwords/br ['stopwords', 'br']
stopwords/ca ['stopwords', 'ca']
stopwords/cs ['stopwords', 'cs']
stopwords/da ['stopwords', 'da']
stopwords/de ['stopwords', 'de']
stopwords/el ['stopwords', 'el']
stopwords/en ['stopwords', 'en']
stopwords/eo ['stopwords', 'eo']
stopwords/es ['stopwords', 'es']
stopwords/et ['stopwords', 'et']
stopwords/eu ['stopwords', 'eu']
stopwords/fa ['stopwords', 'fa']
stopwords/fi ['stopwords', 'fi']
stopwords/fr ['stopwords', 'fr']
stopwords/ga ['stopwords', 'ga']
stopwords/gl ['stopwords', 'gl']
stopwords/gu ['stopwords', 'gu']
stopwords/ha ['stopwords', 'ha']
stopwords/he ['stopwords', 'he']
stopwords/hi ['stopwords', 'hi']
stopwords/hr ['stopwords', 'hr']
stopwords/hu ['stopwords', 'hu']
stopwords/hy ['stopwords', 'hy']
stopwords/id ['stopwords', 'id']
stopwords/it ['stopwords', 'it']
stopwords/ja ['stopwords', 'ja']
stopwords/ko ['stopwords', 'ko']
stopwords/ku ['stopwords', 'ku']
stopwords/la ['stopwords', 'la']
stopwords/lt ['stopwords', 'lt']
stopwords/lv ['stopwords', 'lv']
stopwords/mr ['stopwords', 'mr']
stopwords/ms ['stopwords', 'ms']
stopwords/nl ['stopwords', 'nl']
stopwords/no ['stopwords', 'no']
stopwords/pl ['stopwords', 'pl']
stopwords/pt ['stopwords', 'pt']
stopwords/ro ['stopwords', 'ro']
stopwords/ru ['stopwords', 'ru']
stopwords/sk ['stopwords', 'sk']
stopwords/sl ['stopwords', 'sl']
stopwords/so ['stopwords', 'so']
stopwords/st ['stopwords', 'st']
stopwords/sv ['stopwords', 'sv']
stopwords/sw ['stopwords', 'sw']
stopwords/th ['stopwords', 'th']
stopwords/tl ['stopwords', 'tl']
stopwords/tr ['stopwords', 'tr']
stopwords/uk ['stopwords', 'uk']
stopwords/ur ['stopwords', 'ur']
stopwords/vi ['stopwords', 'vi']
stopwords/yo ['stopwords', 'yo']
stopwords/zh ['stopwords', 'zh']
stopwords/zu ['stopwords', 'zu']
[99/133] Compiling C++ object test/search_iterator.p/search_iterator.cpp.o
In file included from /usr/local/include/gtest/gtest.h:382,
                 from ../test/search_iterator.cpp:27:
../test/search_iterator.cpp: In member function ‘virtual void {anonymous}::search_iterator_uninitialized_Test::TestBody()’:
../test/search_iterator.cpp:41:23: warning: ‘int zim::SearchIterator::getSize() const’ is deprecated [-Wdeprecated-declarations]
   41 |   ASSERT_EQ(it.getSize(), -1);
      |             ~~~~~~~~~~^~
In file included from ../include/zim/search.h:25,
                 from ../test/search_iterator.cpp:22:
../include/zim/search_iterator.h:59:24: note: declared here
   59 |         DEPRECATED int getSize() const;
      |                        ^~~~~~~
In file included from /usr/local/include/gtest/gtest.h:382,
                 from ../test/search_iterator.cpp:27:
../test/search_iterator.cpp:41:23: warning: ‘int zim::SearchIterator::getSize() const’ is deprecated [-Wdeprecated-declarations]
   41 |   ASSERT_EQ(it.getSize(), -1);
      |             ~~~~~~~~~~^~
In file included from ../include/zim/search.h:25,
                 from ../test/search_iterator.cpp:22:
../include/zim/search_iterator.h:59:24: note: declared here
   59 |         DEPRECATED int getSize() const;
      |                        ^~~~~~~
In file included from /usr/local/include/gtest/gtest.h:382,
                 from ../test/search_iterator.cpp:27:
../test/search_iterator.cpp: In member function ‘virtual void {anonymous}::search_iterator_end_Test::TestBody()’:
../test/search_iterator.cpp:67:23: warning: ‘int zim::SearchIterator::getSize() const’ is deprecated [-Wdeprecated-declarations]
   67 |   ASSERT_EQ(it.getSize(), -1);
      |             ~~~~~~~~~~^~
In file included from ../include/zim/search.h:25,
                 from ../test/search_iterator.cpp:22:
../include/zim/search_iterator.h:59:24: note: declared here
   59 |         DEPRECATED int getSize() const;
      |                        ^~~~~~~
In file included from /usr/local/include/gtest/gtest.h:382,
                 from ../test/search_iterator.cpp:27:
../test/search_iterator.cpp:67:23: warning: ‘int zim::SearchIterator::getSize() const’ is deprecated [-Wdeprecated-declarations]
   67 |   ASSERT_EQ(it.getSize(), -1);
      |             ~~~~~~~~~~^~
In file included from ../include/zim/search.h:25,
                 from ../test/search_iterator.cpp:22:
../include/zim/search_iterator.h:59:24: note: declared here
   59 |         DEPRECATED int getSize() const;
      |                        ^~~~~~~
In file included from /usr/local/include/gtest/gtest.h:382,
                 from ../test/search_iterator.cpp:27:
../test/search_iterator.cpp: In member function ‘virtual void {anonymous}::search_iterator_functions_Test::TestBody()’:
../test/search_iterator.cpp:119:23: warning: ‘int zim::SearchIterator::getSize() const’ is deprecated [-Wdeprecated-declarations]
  119 |   ASSERT_EQ(it.getSize(), -1);                 // Unimplemented
      |             ~~~~~~~~~~^~
In file included from ../include/zim/search.h:25,
                 from ../test/search_iterator.cpp:22:
../include/zim/search_iterator.h:59:24: note: declared here
   59 |         DEPRECATED int getSize() const;
      |                        ^~~~~~~
In file included from /usr/local/include/gtest/gtest.h:382,
                 from ../test/search_iterator.cpp:27:
../test/search_iterator.cpp:119:23: warning: ‘int zim::SearchIterator::getSize() const’ is deprecated [-Wdeprecated-declarations]
  119 |   ASSERT_EQ(it.getSize(), -1);                 // Unimplemented
      |             ~~~~~~~~~~^~
In file included from ../include/zim/search.h:25,
                 from ../test/search_iterator.cpp:22:
../include/zim/search_iterator.h:59:24: note: declared here
   59 |         DEPRECATED int getSize() const;
      |                        ^~~~~~~
[133/133] Linking target test/defaultIndexdata
```

--- a/test/search_iterator.cpp
+++ b/test/search_iterator.cpp
@@ -38,7 +38,6 @@ TEST(search_iterator, uninitialized) {
   ASSERT_EQ(it.getScore(), 0);
   ASSERT_EQ(it.getFileIndex(), 0);
   ASSERT_EQ(it.getWordCount(), -1);
-  ASSERT_EQ(it.getSize(), -1);
   ASSERT_THROW(it.getZimId(), std::runtime_error);
   ASSERT_THROW(*it, std::runtime_error);
   ASSERT_THROW(it.operator->(), std::runtime_error);
@@ -64,7 +63,6 @@ TEST(search_iterator, end) {
   ASSERT_THROW(it.getScore(), std::runtime_error);
   ASSERT_THROW(it.getFileIndex(), std::runtime_error);
   ASSERT_THROW(it.getWordCount(), std::runtime_error);
-  ASSERT_EQ(it.getSize(), -1);
   ASSERT_THROW(*it, std::runtime_error);
   ASSERT_THROW(it.operator->(), std::runtime_error);
 }
@@ -116,7 +114,6 @@ TEST(search_iterator, functions) {
   ASSERT_EQ(it.getFileIndex(), 0);
   ASSERT_EQ(it.getZimId(), archive.getUuid());
   ASSERT_EQ(it.getWordCount(), 3);
-  ASSERT_EQ(it.getSize(), -1);                 // Unimplemented
 
   // Check getTitle for accents/cased text
   it++;

